tests/expected/function-contract/type_annotation_needed.rs

```diff
@@ -0,0 +1,14 @@
+// Copyright Kani Contributors
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+// kani-flags: -Zfunction-contracts
+
+#[kani::ensures(result.is_some())]
+fn or_default<T: Default + kani::Arbitrary>(opt: Option<T>) -> Option<T> {
```

Comment:
Can you please add this condition as part of the macro? I.e.: The return type has to implement Arbitrary.